dipy/nn/histo_resdnn.py

```diff
@@ -0,0 +1,245 @@
+            P., Hainline, A. E., Bermudez, C., ... & Stępniewska, I. (2019).
+            Deep learning captures more accurate diffusion fiber orientations
+            distributions than constrained spherical deconvolution.
+            arXiv preprint arXiv:1911.07927.
+        """
+
+        if not have_tf:
+            raise tf()
+
+        self.sh_order = sh_order
+        self.sh_size = len(sph_harm_ind_list(sh_order)[0])
+        self.basis_type = basis_type
+
+        log_level = 'INFO' if verbose else 'CRITICAL'
+        set_logger_level(log_level)
+        if self.basis_type != 'tournier07':
+            logger.warning('Be careful, original weights were obtained '
+                           'from training on the tournier07 basis, '
+                           'unless you re-trained the network, do not '
+                           'change basis!')
+
+        # ResDNN Network Flow
+        num_hidden = self.sh_size
+        inputs = Input(shape=(self.sh_size,))
+        x1 = Dense(400, activation='relu')(inputs)
+        x2 = Dense(num_hidden, activation='relu')(x1)
+        x3 = Dense(200, activation='relu')(x2)
+        x4 = Dense(num_hidden, activation='linear')(x3)
+        res_add = Add()([x2, x4])
+        x5 = Dense(200, activation='relu')(res_add)
+        x6 = Dense(num_hidden)(x5)
+
+        self.model = Model(inputs=inputs, outputs=x6)
+
+    def load_model_weights(self, weights_path):
+        r"""
+        Load the model pre-training weights to use for the fitting.
+        Will not work if the declared SH_ORDER does not match the weights
+        expected input.
+
+        The weights for a sh_order of 8 can be obtained via the function:
+            get_fnames('histo_resdnn_weights').
+
+        Parameters
+        ----------
+        weights_path : str
+            Path to the file containing the weights (hdf5, saved by tensorflow)
+        """
+        try:
+            self.model.load_weights(weights_path)
+        except ValueError:
+            raise ValueError('Expected input for the provided model weights '
+                             'do not match the declared model ({})'
+                             .format(self.sh_size))
+
+    def predict(self, x_test):
+        r"""
+        Predict fODF (as SH) from input raw DWI signal (as SH)
+
+        Parameters
+        ----------
+        x_test : np.ndarray
+            Array of size (N, M) where M is
+            ``(sh_order + 1) * (sh_order + 2) / 2``.
+            N should not be too big as to limit memory usage.
+
+        Returns
+        -------
+        _ : np.ndarray (N, M)
+            Predicted fODF (as SH)
+        """
+
+        if x_test.shape[-1] != self.sh_size:
+            raise ValueError('Expected input for the provided model weights '
+                             'do not match the declared model ({})'
+                             .format(self.sh_size))
+
+        return self.model.predict(x_test)
+
+    def fit(self, data, gtab, mask=None):
```

Comment:
This needs some thinking. I understand what you are trying to do but there is a clear conflict with `fit` function in the model of this module (`dipy.nn.model`). Look [here](https://github.com/dipy/dipy/blob/master/dipy/nn/model.py#L74) 

I know you are in the middle of both worlds (reconstruction vs neural network). However, in our current architecture, the function name  `fit` has two different goals here. 

We need to put some thinking to fix this issue.

Reply:
I can change predict() to __predict() et rename fit() to predict()?

The current predict function could (should) be considered private anyway, so this could work nicely.

Reply:
sounds like a good plan